lib/SPIRV/SPIRVInternal.h

```diff
@@ -774,6 +774,9 @@ int getMDOperandAsInt(MDNode *N, unsigned I);
 /// Get metadata operand as string.
 std::string getMDOperandAsString(MDNode *N, unsigned I);
 
+/// Get metadata operand as another metadata node
+MDNode *getMDOperandAsMDNode(MDNode *N, unsigned I);
```

Comment:
It would be nice to move common improvements into additional commit to separate them from ivdep feature functionality. But it is optional.

Reply:
I can do this to simplify the reviewers' life. Would this include the transition of LoopControl-specific functions into `SPIRVWriter`?

Reply:
As all commits are squashed before to be merged then no more sense in this change. Drop it.

Reply:
Not all commits are squashed. In this repo we are trying to maintain linear history. So normally we push `Rebase and Merge` button which adds all commits from the PR on top of the target branch(master).